### ISSUE TITLE
Crash on my CLI after a certain count of cmd call

### ISSUE THREAD

**Issue by GhomKrosmonaute** [oven-sh/bun#15174]
### How can we reproduce the crash?

![Image](https://github.com/user-attachments/assets/3db21576-665f-4ae9-9f48-457ff0e6e1fd)

I used Bun to run my CLI and after a certain count of commands, not according to the cmd order, Bun crashes.
CLI's technologies:
- npm:inquirer
- npm:commander

### Relevant log output

```shell
============================================================
Bun v1.1.34 (5e5e7c60) Windows x64
Windows v.win10_fe
CPU: sse42 avx avx2
Args: "C:\Users\camil\.bun\bin\bun.exe" "run" "../src/index.ts" "add" "table"
Features: Bun.stdin(2) Bun.stdout dotenv jsc transpiler_cache(4) tsconfig(4) tsconfig_paths(3) 
Builtins: "bun:main" "node:assert" "node:async_hooks" "node:buffer" "node:child_process" "node:crypto" "node:events" "node:fs" "node:fs/promises" "node:os" "node:path" "node:process" "node:readline" "node:stream" "node:string_decoder" "node:tty" "node:url" "node:util" "node:util/types" 
Elapsed: 36496ms | User: 203ms | Sys: 15ms
RSS: 0.22GB | Peak: 0.22GB | Commit: 0.30GB | Faults: 54612

panic(main thread): Segmentation fault at address 0x5
oh no: Bun has crashed. This indicates a bug in Bun, not your code.
```

### Stack Trace (bun.report)

Bun v1.1.34 ([`5e5e7c6`](<https://github.com/oven-sh/bun/tree/5e5e7c60f1004c79a3f87e1f5470ee19d57f10d0>)) on windows x86_64 [AutoCommand]

Segmentation fault at address 0x00000005

- *10 unknown/js code*

Features: Bun.stderr, Bun.stdin, Bun.stdout, fetch, jsc, spawn, transpiler\_cache, tsconfig, tsconfig\_paths, tsconfig\_paths

<sub>Sentry Issue: <strong><a href="https://bun-p9.sentry.io/issues/5298323857/">BUN-85</a></strong></sub>

**Comment by nektro** [comment]
may be a duplicate of https://github.com/oven-sh/bun/issues/10844

**Comment by heimskr** [comment]
Probably fixed by #15357.
